yjit.rb

```diff
@@ -247,10 +248,30 @@ def self.simulate_oom! # :nodoc:
     at_exit { print_and_dump_stats }
   end
 
+  # Blocks that are called when YJIT is enabled
+  @yjit_hooks = []
+
   class << self
     # :stopdoc:
     private
 
+    # Register a block to be called when YJIT is enabled
+    def add_yjit_hook(hook)
+      @yjit_hooks << hook
+    end
+
+    # Run YJIT hooks registered by RubyVM::YJIT.with_yjit
+    def call_yjit_hooks
+      # Skip using builtin methods in Ruby if --yjit-c-builtin is given
+      return if Primitive.yjit_c_builtin_p
+
+      # Unset GET_VM()->running to give METHOD_ENTRY_BASIC flag to methods defined by the hooks
+      Primitive.cexpr! 'GET_VM()->running = 0'
```

Comment:
This seems no thread safe. It's probably fine without ractors, but theoretically we could get a timed thread switch in the middle of this region with plain `Thread`s too, then this would at least allow potentially disguising methods as basic. I don't know what other special privilege running the this flag cleared grants.

Reply:
How about this? https://github.com/ruby/ruby/pull/11955/commits/d066d0173ae0cd268c33c1a7355aa0e08700351d

Reply:
Thanks! Sorry for bugging you about it.